src/StoreKeeper/WooCommerce/B2C/Cron/CronRegistrar.php

```diff
@@ -0,0 +1,156 @@
+    public const RUNNER_CRONTAB_CLI = 'crontab-cli';
+    public const CRONTAB_RUNNERS = [
+        self::RUNNER_CRONTAB_CLI,
+        self::RUNNER_CRONTAB_API,
+    ];
+
+    public const STATUS_UNEXECUTED = 'unexecuted';
+    public const STATUS_FAILED = 'failed';
+    public const STATUS_SUCCESS = 'success';
+
+    public function addCustomCronInterval($schedules): array
+    {
+        $schedules[self::SCHEDULES_EVERY_MINUTE_KEY] = [
+            'interval' => 60,
+            'display' => __('Every minute', I18N::DOMAIN),
+        ];
+
+        return $schedules;
+    }
+
+    public function register(): void
+    {
+        $runner = CronOptions::get(CronOptions::RUNNER, self::RUNNER_WPCRON);
+        if (self::RUNNER_WPCRON === $runner && !wp_next_scheduled(self::HOOK_PROCESS_TASK)) {
+            wp_schedule_event(time(), self::SCHEDULES_EVERY_MINUTE_KEY, self::HOOK_PROCESS_TASK);
+        }
+    }
+
+    /**
+     * @throws CronRunnerException
+     */
+    public static function checkRunnerStatus(): void
+    {
+        $runner = CronOptions::get(CronOptions::RUNNER);
+
+        if (self::RUNNER_WPCRON === $runner) {
+            static::checkWpCronStatus();
+        }
+    }
+
+    /**
+     * Forked from WP Crontrol v1.10.0.
+     *
+     * @see https://wordpress.org/plugins/wp-crontrol test_cron_spawn()
+     *
+     * @throws CronRunnerException
+     */
+    private static function checkWpCronStatus(): void
+    {
+        global $wp_version;
+
+        $sslVerify = version_compare($wp_version, 4.0, '<');
+        $lockTime = sprintf('%.22F', microtime(true));
+
+        $cronRequest = apply_filters('cron_request', [
+            'url' => add_query_arg('doing_wp_cron', $lockTime, site_url('wp-cron.php')),
+            'key' => $lockTime,
+            'args' => [
+                'timeout' => 3,
+                'blocking' => true,
+                'sslverify' => apply_filters('https_local_ssl_verify', $sslVerify),
+            ],
+        ]);
+
+        $cronResponse = wp_remote_post($cronRequest['url'], $cronRequest['args']);
+
+        if (is_wp_error($cronResponse)) {
+            throw new CronRunnerException(sprintf('WP Cron encountered an error and may not work: %s', I18N::DOMAIN), strip_tags($cronResponse->get_error_message()));
+        } elseif (wp_remote_retrieve_response_code($cronResponse) >= 300) {
+            throw new CronRunnerException(sprintf(__('WP Cron return an unexpected HTTP response code: %s', I18N::DOMAIN), (int) wp_remote_retrieve_response_code($cronResponse)));
+        }
+    }
+
+    public static function buildMessage(): array
+    {
+        $runner = CronOptions::get(CronOptions::RUNNER);
+
+        $message = __(
+            'Cron may not be running:',
+            I18N::DOMAIN
+        );
+
+        $description = __('Contact your system administrator if the problem persists', I18N::DOMAIN);
+
+        if (in_array($runner, self::CRONTAB_RUNNERS, true)) {
```

Comment:
maybe we can just put link here to the scheduler page? Cos there are all the checks and how to configure it

Reply:
Added link to scheduler settings tab instead.